src/managedwriter/writer_client.ts

```diff
@@ -0,0 +1,311 @@
+    this._client = client;
+  };
+
+  getWriteStreams = (writeStream: WriteStream): undefined | null | string[] => {
+    if (writeStream === undefined || writeStream.name === undefined) {
+      return undefined;
+    }
+    if (writeStream.name === null) {
+      return null;
+    }
+    return new Array(writeStream.name);
+  };
+
+  setWriteStreamType(streamType: WriteStream['type']): void {
+    this._writeStreamType = streamType;
+  }
+
+  getWriteStreamType(): WriteStream['type'] {
+    return this._writeStreamType;
+  }
+
+  getConnections(): StreamConnections {
+    return this._connections;
+  }
+
+  getClientClosedStatus(): boolean {
+    return this._client_closed;
+  }
+
+  async initializeStreamConnection(clientOptions?: CallOptions): Promise<void> {
+    if (this._client_closed) {
+      this._client_closed = false;
+    }
+    const request: protos.google.cloud.bigquery.storage.v1.ICreateWriteStreamRequest =
+      {
+        parent: this.getParent(),
+        writeStream: {type: this.getWriteStreamType()},
+      };
+    const [response] = await this._client.createWriteStream(request);
+    if (typeof [response] === undefined) {
+      throw new gax.GoogleError(`${response}`);
+    }
+    console.log(`Stream connection created: ${response.name}`);
+    try {
+      if (response.name) {
+        this.setStreamId(response.name);
+        const write_stream: WriteStream = {
+          name: response.name,
+          type: this.getWriteStreamType(),
+        };
+        const stream_connection: StreamConnection = {
+          write_stream: write_stream,
+          connection: clientOptions
+            ? this._client.appendRows(clientOptions)
+            : this._client.appendRows(),
+        };
+        this._connections.connection_list.push(stream_connection);
+        this._connections.connections[`${write_stream.name}`] =
+          stream_connection.connection;
+      }
+    } catch {
+      throw new Error('Stream connection failed');
+    }
+  }
+
+  async appendRowsToStream(
+    streamConnection: WriteStream['name'],
+    serializedRows: ProtoData,
+    offsetValue: IInt64Value
+  ): Promise<AppendRowResponse[]> {
+    const responses: AppendRowResponse[] | null = [];
+    const connection: gax.CancellableStream =
+      this._connections.connections[`${streamConnection}`];
+    const request: AppendRowRequest = {
+      writeStream: streamConnection,
+      protoRows: serializedRows,
+      offset: offsetValue,
+    };
+
+    connection.write(request, () => {
+      connection.on('data', response => {
+        if (response.error) {
+          throw new Error(response.error.message);
+        }
+
+        responses.push(response);
+
+        if (responses.length === 1) {
+          connection.end(() => {
+            this.closeStream();
+          });
+        }
+      });
+
+      connection.on('error', err => {
+        throw err;
+      });
+    });
+    return responses;
+  }
+
+  async closeStream(): Promise<void> {
```

Comment:
Closing the connection/stream and committing/finalizing the stream should be separated actions, specially because for Default Stream, they are a Stream of type `COMMITED` and doesn't support calling `finalizeWriteStream`, but users would also want to close the connection to end the streaming. For user created WriteStreams of type `PENDING`, they will call finalize to commit the changes and then close the connection later.